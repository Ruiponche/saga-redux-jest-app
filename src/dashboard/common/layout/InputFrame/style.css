.highlight {
  outline: 1px solid #dddddd;
}
