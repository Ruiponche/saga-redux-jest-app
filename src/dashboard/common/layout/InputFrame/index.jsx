import React, { Component } from 'react'
import './style.css'

class InputFrame extends Component {
  constructor(props) {
    super(props)
    this.state = { className: 'normal frame' }
  }

  render() {
    const { className } = this.state

    return (
      <div
        className={className}
        onMouseEnter={() => this.setState({ className: 'highlight frame' })}
        onMouseLeave={() => this.setState({ className: 'normal frame' })}
      >
        {this.props.children}
      </div>
    )
  }
}

export default InputFrame
