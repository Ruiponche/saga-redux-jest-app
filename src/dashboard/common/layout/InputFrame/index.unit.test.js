import React from 'react'
import renderer from 'react-test-renderer'
import InputFrame from './index'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

const child = <div>Sample text</div>

configure({ adapter: new Adapter() })

test('input frame class changes when hovered', () => {
  const inputFrameComponent = renderer.create(<InputFrame>{child}</InputFrame>)
  let tree = inputFrameComponent.toJSON()
  expect(tree).toMatchSnapshot()

  tree.props.onMouseEnter()
  tree = inputFrameComponent.toJSON()
  expect(tree).toMatchSnapshot()

  tree.props.onMouseLeave()
  tree = inputFrameComponent.toJSON()
  expect(tree).toMatchSnapshot()
})

describe('<InputFrame/>', () => {
  it('should match shallowed component', () => {
    const wrapper = shallow(<InputFrame />)
    expect(wrapper).toMatchSnapshot()
  })

  it("should contain a 'normal' in its className", () => {
    const wrapper = shallow(<InputFrame />)
    expect(wrapper.find('.frame').hasClass('normal')).toBe(true)
  })

  it("should not contain a 'highlight' in its className", () => {
    const wrapper = shallow(<InputFrame />)
    expect(wrapper.find('.frame').hasClass('highlight')).toBe(false)
  })

  it("on mouse enter 'normal' -> 'highlight, on mouse leave  'highlight' -> 'normal' ", () => {
    const wrapper = shallow(<InputFrame />)
    wrapper.find('.frame').simulate('mouseenter')
    expect(wrapper.find('.frame').hasClass('highlight')).toBe(true)

    wrapper.find('.frame').simulate('mouseleave')
    expect(wrapper.find('.frame').hasClass('highlight')).toBe(false)
    expect(wrapper.find('.frame').hasClass('normal')).toBe(true)
  })
})
