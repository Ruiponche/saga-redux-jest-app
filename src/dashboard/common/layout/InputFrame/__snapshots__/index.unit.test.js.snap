// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputFrame/> should match shallowed component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputFrame />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "normal frame",
      "onMouseEnter": [Function],
      "onMouseLeave": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "normal frame",
        "onMouseEnter": [Function],
        "onMouseLeave": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`input frame class changes when hovered 1`] = `
<div
  className="normal frame"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div>
    Sample text
  </div>
</div>
`;

exports[`input frame class changes when hovered 2`] = `
<div
  className="highlight frame"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div>
    Sample text
  </div>
</div>
`;

exports[`input frame class changes when hovered 3`] = `
<div
  className="normal frame"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div>
    Sample text
  </div>
</div>
`;
